cmake-modules/AddTestCMocka.cmake

```diff
@@ -77,7 +77,21 @@ function(ADD_CMOCKA_TEST _TARGET_NAME)
     endif()
 
     if(DEFINED ENV{VCPKG_ROOT} OR DEFINED ENV{VCPKG_INSTALLATION_ROOT})
-        set(CMOCKA_LIB ${CMOCKA_LIBRARIES})
+        if(DEFINED ENV{VCPKG_ROOT})
+            set(VCPKG_PATH $ENV{VCPKG_ROOT})
+        endif()
+        if(DEFINED ENV{VCPKG_INSTALLATION_ROOT})
+            set(VCPKG_PATH "$ENV{VCPKG_INSTALLATION_ROOT}/")
+        endif()
+
+        # Patch for new VCPKG cmocka returning relative path
+        if(MSVC)
+            set(CMOCKA_STATIC_LIB "/lib/cmocka-static.lib")
+        else()
+            set(CMOCKA_STATIC_LIB "/lib/libcmocka-static.a")
```

Comment:
Question: Is the name of the `.a` file accurate? Is it supposed to be `libcmocka-static`?

Reply:
it is .a on Posix (lin/mac) and .lib on windows.

the expectation is that cmoka will use whatever is loaded on ${CMOCKA_LIBRARIES}.  That variable is provided by VCPKG when cmocka is installed.   Cmocka provides the scripts to VCPKG to generate this variable.

In the past, this variable was generated like:
`degug;/path/to/VCPKG/{vcpkg packages installed path}/libraries;optimized;/path/to/VCPKG/{vcpkg packages installed path}/libraries`
but cmocka made a change and now it looks like:
`debug;/libraries;optimized;/libraries`

So the prefix path to the libraries was removed from both, debug and optimized libs.

I am just manually building the value we used to have before